BuildResidentialHPXML/measure.rb

```diff
@@ -4142,33 +4165,59 @@ def self.set_hvac_distribution(hpxml, runner, args)
         end
       end
     end
-    return unless air_distribution_systems.size > 0
+
+    # FanCoil?
+    fan_coil_distribution_systems = []
+    hpxml.heating_systems.each do |heating_system|
+      if args[:heating_system_type].include?('Fan Coil')
+        fan_coil_distribution_systems << heating_system
+      end
+    end
+
+    return if air_distribution_systems.size == 0 && fan_coil_distribution_systems.size == 0
 
     if args[:ducts_number_of_return_registers] != Constants.Auto
       number_of_return_registers = args[:ducts_number_of_return_registers]
     end
 
-    hpxml.hvac_distributions.add(id: 'AirDistribution',
-                                 distribution_system_type: HPXML::HVACDistributionTypeAir,
-                                 conditioned_floor_area_served: args[:geometry_cfa],
-                                 number_of_return_registers: number_of_return_registers)
-
-    air_distribution_systems.each do |hvac_system|
-      hvac_system.distribution_system_idref = hpxml.hvac_distributions[-1].id
+    if air_distribution_systems.size > 0
+      hpxml.hvac_distributions.add(id: 'AirDistribution',
+                                   distribution_system_type: HPXML::HVACDistributionTypeAir,
+                                   conditioned_floor_area_served: args[:geometry_cfa],
+                                   number_of_return_registers: number_of_return_registers)
+      air_distribution_systems.each do |hvac_system|
+        hvac_system.distribution_system_idref = hpxml.hvac_distributions[-1].id
+      end
+      set_duct_leakages(args, hpxml.hvac_distributions[-1])
+      set_ducts(args, hpxml.hvac_distributions[-1])
+    end
+
+    if fan_coil_distribution_systems.size > 0
+      hpxml.hvac_distributions.add(id: 'FanCoilDistribution',
+                                   distribution_system_type: HPXML::HVACDistributionTypeAir,
+                                   conditioned_floor_area_served: args[:geometry_cfa],
+                                   number_of_return_registers: number_of_return_registers,
+                                   air_type: HPXML::AirTypeFanCoil)
+      fan_coil_distribution_systems.each do |hvac_system|
+        hvac_system.distribution_system_idref = hpxml.hvac_distributions[-1].id
+      end
+      set_duct_leakages(args, hpxml.hvac_distributions[-1])
```

Comment:
This isn't correct, as it's going to assign duct leakage to the fan coil unit, but we want the fan coil unit to be ductless. Instead just assign values of zero for supply and return leakage. (This reminds me that I should really update OS-HPXML so that duct leakages are not required for fan coils...)

Also, I don't understand how your base-bldgtype-multifamily-shared-boiler-only-fan-coil.osw passed the tests? The HPXML it is compared to clearly has zero duct leakage for the ductless fan coil's distribution system. Can you explain that?

Reply:
The test is based on `base-bldgtype-multifamily.osw`, which has values of zero for supply and return leakage.